primitives/session-manager/src/pallet_session_support.rs

```diff
@@ -0,0 +1,108 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::marker::PhantomData;
+use derive_new::new;
+use frame_system::pallet_prelude::BlockNumberFor;
+use log::debug;
+use sp_staking::SessionIndex;
+use sp_std::vec::Vec;
+
+/// Implements [`pallet_session::SessionManager`] and [`pallet_session::ShouldEndSession`] integrated with [`pallet_session_validator_management`].
+///
+/// To use it, wire it in runtime configuration of [`pallet_session`].
+#[allow(dead_code)]
+#[derive(new)]
+pub struct PalletSessionSupport<T> {
+	_phantom: PhantomData<T>,
+}
+
+impl<T: pallet_session_validator_management::Config + pallet_session::Config>
+	pallet_session::SessionManager<T::AccountId> for PalletSessionSupport<T>
+{
+	/// Sets the first validator-set by mapping the current committee from [`pallet_session_validator_management`]
+	fn new_session_genesis(_new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+		Some(
+			pallet_session_validator_management::Pallet::<T>::current_committee_storage()
+				.committee
+				.into_iter()
+				.map(|(id, _)| id.into())
+				.collect::<Vec<_>>(),
+		)
+	}
+
+	/// Rotates the committee in [`pallet_session_validator_management`] and plans this new committee as upcoming validator-set.
+	/// Updates the session index of [`pallet_session`].
+	// Instead of Some((*).expect) we could just use (*). However, we rather panic in presence of important programming errors.
+	fn new_session(new_index: SessionIndex) -> Option<Vec<T::AccountId>> {
+		debug!("PalletSessionSupport: New session {new_index}");
+		pallet_session::pallet::CurrentIndex::<T>::put(new_index);
+		Some(
+			pallet_session_validator_management::Pallet::<T>::rotate_committee_to_next_epoch()
+				.expect(
+					"PalletSessionSupport: Session should never end without current epoch validators defined. \
+				Check ShouldEndSession implementation or if it is used before starting new session",
```

Comment:
Hard to understand wording imo. How about this?

```suggestion
				This may be caused by ShouldEndSession containing an error or being used before starting new session",
```